### ISSUE TITLE
New compiler: parsing mistake involving incorrect increment/decrement operators

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2019]
CC @fernewelten 
Consider script:

```
int a = 10;
int b = 5;
int c = a ++ b;
    
Display("a = %d, b = %d, c = %d", a, b, c);
```

The output is:
> a = 10, b = 5, c = 13 (???)

What is expected is probably a script error, as the above statement does not have a correct interpretation.

**Comment by fernewelten** [comment]
That's a bug. `++` is mis-parsed as a binary operator. I'll look into it.

**Comment by fernewelten** [comment]
Addressed in #2038.

**Comment by ivan-mogilko** [comment]
Fixed by #2038
